test/functional/test_framework/util.py

```diff
@@ -524,6 +524,10 @@ def fill_mempool(test_framework, node, miniwallet):
 
     test_framework.log.debug("Create a mempool tx that will be evicted")
     tx_to_be_evicted_id = miniwallet.send_self_transfer(from_node=node, fee_rate=relayfee)["txid"]
+    # remove the newly created coin from MiniWallet's UTXO set, in order to avoid it being spent
+    # in one of the mempool filling txs below (which could bump up its descendant fee-rate and
+    # hence stop it from being evicted)
+    miniwallet.get_utxo(txid=tx_to_be_evicted_id, mark_as_spent=True)
```

Comment:
Nice catch! I wonder if a more wholistic way to avoid unintentionally creating tx packages is to gather all the (confirmed only) UTXOs at the beginning? e.g. [this](https://github.com/glozow/bitcoin/commits/2024-04-fill-mempool-29827/), but feel free to ignore